source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,218 @@
+﻿using System;
+using System.Collections.Generic;
+//using Il2CppSystem.Collections.Generic;
+using System.Text;
+using UnityEngine;
+using Reactor;
+using TownOfUs.Roles;
+using TownOfUs.Roles.Modifiers;
+using System.Linq;
+using HarmonyLib;
+
+namespace TownOfUs.Patches
+{
+    [HarmonyPatch]
+    static class EndGameSummary
+    {
+        static class AdditionalTempData
+        {
+            public static List<PlayerInfo> playerData = new List<PlayerInfo>();
+
+            public static void clear()
+            {
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public int PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static System.Collections.Generic.List<PlayerControl> allPlayers = new System.Collections.Generic.List<PlayerControl>();
+        public static bool playerInfoUpdateFlag1Updating = false;
+        public static bool playerInfoUpdateFlag2EndOfGame = false;
+
+        public static void UpdatePlayerDataRPCCall()
+        {
+            var writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
+        (byte)CustomRPC.UpdateGamePlayerControlData, Hazel.SendOption.Reliable, -1);
+            AmongUsClient.Instance.FinishRpcImmediately(writer);
+        }
+
+        public static void GatherPlayerData(byte rpccallid)
+        {
+            EndGameSummary.allPlayers.Clear();
+            EndGameSummary.allPlayers = EndGameSummary.GetAllPlayers();
+            List<PlayerControl> allplayersList = EndGameSummary.allPlayers;
+            UpdatePlayerInfo(allplayersList);
+        }
+
+        public static void UpdatePlayerInfo(List<PlayerControl> players)
+        {
+            List<AdditionalTempData.PlayerInfo> pinfolist = new List<AdditionalTempData.PlayerInfo>();
+            pinfolist.Clear();
```

Comment:
Why `Clear()` when it is a newly created list?
```suggestion
            var pinfolist = new List<AdditionalTempData.PlayerInfo>();
```

Reply:
no thats fair, ive been juggling with various ways of doing things, i thought i cleaned up the code, but i guess i left that in ;)